### PR TITLE
CCP-1822: Initial implementation

### PR DESCRIPTION
**Pull request by dqdquint**
Implemented a configurable video/image widget.

Set up a test widget in shop no: 31887.

Had a couple of different problems when implementing this feature.
- I tried used the media slider we have in engage but that proved to be unsuccessful since media is only gained if it is on the product that is passed.
- I tried using the MediaVideo component within the slider i created in the extension but this proved to be buggy in testing with looped videos not rendering on the first loop.
- I opted to go with using img and iframe tags to render slides and this works, except with auto play it only turns off when interacting with the next or prev buttons. I tried setting a ref to the iframe, but was unsuccessful in getting a click in the iframe to pause auto play in the slider. There is a method on the slider that we can call to halt auto play. 
- In testing this auto play issue didn't prove to be critical because when a user plays a video the video opens out of the slider in full screen.

### CODE REVIEW

**Review by Carsten89 — CHANGES_REQUESTED** [review]
(this is not a real review)
Please add:
- support for german lang
- /demo folder

**Comment by aaron-martin** [comment]
Custom is misspelled in the description of the ConfigurableCarousel widget

**Comment by aaron-martin** [comment]
the alignment on the buttons seem off to me. I think they should be vertically aligned middle within the gray area. This is only a problem in the Android app because it shows the gray area.
image

**Comment by aaron-martin** [comment]
On android the videos don't open in full screen so it is possible to slide away from the video while it is playing.

File: frontend/components/VideoCarousel/index.jsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    spaceBetween: spaceBetweenSlides,
+    slidesPerView: numberSlidesPerView,
+    loop: true,
+    autoplay: autoplayValue,
+    noSwipeing: true,
+    getSwiper: updateSwiper,
+  };
+
+  const slides = slideInfo.map((slide, index) => {
+    const item = slide.type === 'video' ?
```
Comment (aaron-martin) [inline frontend/components/VideoCarousel/index.jsx]: I am not convinced that a ternary expression is the most readable choice here. I think this is more readable. 

```
  const slides = slideInfo.map((slide, index) => {
    if (slide.type === 'video') {
      return (
        <div
          className={styles.videoWrapper(backgroundColor)}
          key={index.toString()}
        >
          <div className={styles.videoResponsive}>
            <iframe className="swiper-lazy" title={slide.alt} data-src={slide.src}/>
            <div className="swiper-lazy-preloader swiper-lazy-preloader-black"/>
          </div>
        </div>
      );
    }
    return (
      <div
        key={index.toString()}
      >
        <img
          src={slide.src}
          alt={slide.alt}
        />
      </div>
    );
  });
```

**Review by aaron-martin — CHANGES_REQUESTED** [review]

File: frontend/components/VideoCarousel/index.jsx (hunk 1 above)
Comment (dqdquint) [reply frontend/components/VideoCarousel/index.jsx]: Updated thanks!

**Comment by dqdquint** [comment]
@aaron-martin  I added additional margin to the carousel buttons, corrected the typo in ConfigurableCarousel , and disabled autoplay to address the video play issue after I confirmed that was reasonable with veronica.

File: demo/index.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+```
+
+![](ButtonConfigurations.png)
+
+### Carousel Configurations
+- `backgroundColor`: Expects a color value as a string. Used to provide a background for loading videos.
+- `numberSlidesPerView`: Expects a number. Sets the number of slides in view
+- `spaceBetweenSlides`: Expects a number. Sets the space between slides
+- `slideInfo`: Expects an array containing information for each slide in the carousel
+  - `alt`: Expects a string that provies a alternative titled for a slide
```
Comment (aaron-martin) [inline demo/index.md]: provides is misspelled

File: demo/index.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+#### Config:
+```json 
+{
+  "settings":{
+    "backgroundColor": "black",
+    "numberSlidesPerView": 1,
+    "spaceBetweenSlides": 30,
+    "slideInfo": [ 
+      { 
+        "alt":"monin 1",
```
Comment (aaron-martin) [inline demo/index.md]: I don't think we should use the Monin name here. Please replace this with something more generic.

**Review by aaron-martin — CHANGES_REQUESTED** [review]
just a few other small changes.

File: demo/index.md (hunk 3 above)
Comment (dqdquint) [reply demo/index.md]: Updated

File: demo/index.md (hunk 2 above)
Comment (dqdquint) [reply demo/index.md]: Updated. Thanks!

**Review by aaron-martin — APPROVED** [review]

**Pull request merged**